tests/testthat/test_fragments2.R

```diff
@@ -0,0 +1,144 @@
+## For additional tests, refer to calculateFragments from PSMatch package
+test_that("calculateFragments2: Default behavior without modifications", {
+    
+    ## Test 1: Default behavior without modifications
+    sequence <- "PQR"
+    original_result <- calculateFragments(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        modifications = NULL,
+        neutralLoss = defaultNeutralLoss(),
+        verbose = FALSE
+    )
+    result <- calculateFragments2(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        fixed_modifications = NULL,
+        variable_modifications = NULL,
+        max_mods = Inf,
+        neutralLoss = defaultNeutralLoss(),
+        verbose = FALSE
+    )
+    
+    ## Check unique peptide without modifications
+    expect_equal(unique(result$peptide), "PQR")
+    expect_equal(result[,-7], original_result)
+    })
+
+test_that("calculateFragments2: Behavior with fixed modifications", {
+    ## Test 2: Fixed modifications
+    sequence <- "PQR"
+    result <- calculateFragments2(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        fixed_modifications = NULL,
+        variable_modifications = NULL,
+        max_mods = Inf,
+        neutralLoss = list(water = c(), ammonia = c()),
+        verbose = FALSE
+    )
+    fixed_modifications <- c(P = 79.966)
+    result_fixed <- calculateFragments2(
+        sequence = sequence,
+        type = c("b", "y"),
+        z = 1,
+        fixed_modifications = fixed_modifications,
+        variable_modifications = NULL,
+        max_mods = 0,
+        neutralLoss = list(water = c(), ammonia = c()),
+        verbose = FALSE
+    )
+    
+    ## Fixed modifications do not produce additional unique peptides
+    expect_equal(length(unique(result_fixed$peptide)), 1)
```

Comment:
May be use the following, that does the same, and also makes sure the sequence is the same.
```
expect_identical(unique(result_fixed$peptide), "PQR")
```